src/account/mnemonic.ts

```diff
@@ -1,9 +1,40 @@
 import { Bee, Reference, Utils } from '@ethersphere/bee-js'
-import { assertBase64UrlData, assertUsername } from './utils'
+import { assertBase64UrlData, assertUsername, removeZeroFromHex } from './utils'
 import { Wallet } from 'ethers'
 import { getFeedData, writeFeedData } from '../feed/api'
 import { Connection } from '../connection/connection'
 import { stringToBytes } from '../utils/bytes'
+import { PublicKey } from '../utils/type'
+import { decrypt } from './encryption'
+import { wrapChunkHelper } from '../feed/utils'
+
+export const ADDRESS_LENGTH = 64
+
+/**
+ * Downloads encrypted mnemonic phrase from swarm chunk
+ *
+ * @param bee Bee client
+ * @param publicKey FDP account public key
+ * @param password FDP account password
+ * @param address FDP account address
+ *
+ * @returns encrypted mnemonic phrase in Base64url format
+ */
+export async function getEncryptedMnemonicByPublicKey(
+  bee: Bee,
+  publicKey: PublicKey,
+  password: string,
+  address: Utils.EthAddress,
+): Promise<string> {
+  const topic = removeZeroFromHex(publicKey) + password
```

Comment:
1) Do you mean hash(publicKey+password)? It converts the topic like this in `getFeedData`.

2) The public key is obtained from ENS in string format and comes here. Do I need to additionally convert a string from ENS to a binary format and accept the binary format in this method?

Reply:
yeah, my comment was related to in which format we pass the parameters to the hashing function. I prefer the byte representation more for public key, because handling as string may lead to confusion and differences (like prefixing with `0x` or not, etc.)